DEV-GUIDE.md

```diff
@@ -1,16 +1,8 @@
 # Developing for Istio
 
-This document is intended to be the canonical source of truth for things like
-supported toolchain versions for building Istio components like Mixer
-(which is used throughout this document as an example when referring to a specific
-component repo or directory).
-If you find a requirement that this doc does not capture, or if you find other
-docs with references to requirements that are not simply links to this doc,
-please [submit an issue](https://github.com/istio/istio/issues/new).
-
-This document is intended to be relative to the branch in which it is found.
-It is guaranteed that requirements will change over time for the development
-branch, but release branches should not change.
+This document helps you get started to develop code for Istio.
```

Comment:
@costinm / @andraxylia after this gets merged, can you please take a crack at this and update it? This is currently all bazel